packages/calcite-components/src/components/block/block.scss

```diff
@@ -23,6 +23,10 @@
   @apply justify-start;
 }
 
+.header--present {
```

Comment:
Nitpick: I believe we use `.<part>--has-text` in other components for similar contexts.